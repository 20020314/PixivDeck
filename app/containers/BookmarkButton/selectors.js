// @flow
import { createSelector } from 'reselect'
import type { State } from 'types/state'

const getSelectIllust = (state: State, { id }: { id: number }) =>
  state.illustById[id]

export const makeSelectIllust = () => createSelector(getSelectIllust, s => s)

export const makeIsBookmarked = () =>
  createSelector(makeSelectIllust(), s => s.isBookmarked)
