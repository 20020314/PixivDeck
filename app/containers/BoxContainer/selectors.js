// @flow
import { createSelector } from 'reselect'
import type { State } from 'types/state'

type Props = {
  id: number,
}

const getSelectUserById = (s: State) => s.userById

const getSelectIllust = (s: State, { id }: Props) => s.illustById[id]

export const makeSelectIllust = () => createSelector(getSelectIllust, s => s)

const makeSelectIllustUserId = () =>
  createSelector(makeSelectIllust(), s => s.user)

export const makeSelectUser = () =>
  createSelector(
    getSelectUserById,
    makeSelectIllustUserId(),
    (users, id) => users[id]
  )

const getSettingModal = (s: State) => s.SettingModal

export const makeSelectIsIllustOnly = () =>
  createSelector(getSettingModal, s => s.isShowCaption)

export const makeSelectIsShowOnlyIllust = () =>
  createSelector(getSettingModal, s => s.isShowOnlyIllust)
