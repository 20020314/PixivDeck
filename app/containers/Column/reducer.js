// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
