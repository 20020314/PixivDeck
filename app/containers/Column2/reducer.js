// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'

type ColumnId = string
type BodyId = string
type BodyType = 'RANKING' | 'BOOKMARK'

export type Column = {
  id: ColumnId,
  title: string,
  bodyId: BodyId,
  type: BodyType,
}

export type Body = {
  id: BodyId,
  type: BodyType,
}

export type Ranking = {
  endpoint: '/v1/illust/ranking',
  params: {
    mode: string,
    offset: number,
  },
} & Body

export type State = {
  [id: string]: Column,
}

const initialState: State = {}

export default function(
  state: State = initialState,
  action: Action
): $Shape<State> {
  switch (action.type) {
    case Actions.ADD_COLUMN:
      return { ...state, [action.id]: action.body }
    default:
      return state
  }
}
