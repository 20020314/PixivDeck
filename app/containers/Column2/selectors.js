// @flow
import { createSelector } from 'reselect'
import type { State } from 'types/state'

type Props = {
  id: string,
}

const getColumn = (state: State, { id }: Props) => state.Column2[id]

export const makeSelectColumn = () => createSelector(getColumn, s => s)
