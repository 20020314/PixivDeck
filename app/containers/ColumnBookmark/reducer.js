// @flow
import { handleRehydrate } from 'util/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Restrict = 'public' | 'private'

export type Endpoint = '/v1/user/bookmarks/illust'

export type ColumnId = Restrict

export type ColumnBookmark = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [ColumnId]: ColumnBookmark }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_BOOKMARK_COLUMN_SUCCESS:
      return { ...state, [action.id]: { illustIds: [], nextUrl: null } }

    case Actions.SET_NEXT_URL: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], nextUrl: action.nextUrl },
      }
    }

    case Actions.FETCH_BOOKMARK_SUCCESS:
    case Actions.FETCH_NEXT_BOOKMARK_SUCCESS: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], illustIds: action.ids },
      }
    }

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnBookmark')
    default:
      return state
  }
}
