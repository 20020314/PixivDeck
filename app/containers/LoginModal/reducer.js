// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'

export type State = {
  username: string,
  password: string,
  isLoading: boolean,
  isLoginFailure: boolean,
}

export const initialState: State = {
  username: '',
  password: '',
  isLoading: false,
  isLoginFailure: false,
}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.SET_AUTH:
      return { ...state, username: action.username, password: action.password }
    case Actions.LOGIN_REQUEST:
      return { ...state, isLoading: true }
    case Actions.START_LOADING:
      return { ...state, isLoading: true }
    case Actions.END_LOADING:
      return { ...state, isLoading: false }
    case Actions.LOGIN_FAILURE:
      return { ...state, isLoginFailure: true }
    case Actions.CLEAR_ERROR:
      return { ...state, isLoginFailure: false }
    default:
      return state
  }
}
