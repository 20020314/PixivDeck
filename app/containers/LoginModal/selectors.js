// @flow
import { createSelector } from 'reselect'
import type { State } from 'types/state'

const selectAuth = (state: State) => state.LoginModal

export const makeSelectUsername = () =>
  createSelector(selectAuth, s => s.username)

export const makeSelectPassword = () =>
  createSelector(selectAuth, s => s.password)

export const makeSelectIsLoading = () =>
  createSelector(selectAuth, s => s.isLoading)

export const makeSelectIsLoginFailure = () =>
  createSelector(selectAuth, s => s.isLoginFailure)
