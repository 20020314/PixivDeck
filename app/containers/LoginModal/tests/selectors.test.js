// @flow
import * as selectors from '../selectors'
import { initialState } from '../reducer'

const mockStore = {
  LoginModal: {
    ...initialState,
    ...{
      username: 'fakeName',
      password: 'fakePassword',
    },
  },
}

test('makeSelectInfo', () => {
  const getInfo = selectors.makeSelectInfo()
  expect(getInfo(mockStore)).toMatchSnapshot()
})

test('makeSelectIsLoginFailure', () => {
  const s = selectors.makeSelectIsLoginFailure()
  expect(s(mockStore)).toBeFalsy()
})

test('', () => {
  const s = selectors.makeSelectIsLoading()
  expect(s(mockStore)).toBeFalsy()
})
