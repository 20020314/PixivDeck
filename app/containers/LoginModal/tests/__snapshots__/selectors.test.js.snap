// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectInfo 1`] = `
Object {
  "password": "fakePassword",
  "username": "fakeName",
}
`;
